mmseg/models/backbones/hrformer.py

```diff
@@ -0,0 +1,1062 @@
+    def _make_transition_layer(self, num_channels_pre_layer,
+                               num_channels_cur_layer):
+        """Build transition layers."""
+        num_branches_cur = len(num_channels_cur_layer)
+        num_branches_pre = len(num_channels_pre_layer)
+
+        transition_layers = []
+        for i in range(num_branches_cur):
+            if i < num_branches_pre:
+                if num_channels_cur_layer[i] != num_channels_pre_layer[i]:
+                    transition_layers.append(
+                        nn.Sequential(
+                            build_conv_layer(
+                                self.conv_cfg,
+                                num_channels_pre_layer[i],
+                                num_channels_cur_layer[i],
+                                3,
+                                1,
+                                1,
+                                bias=False,
+                            ),
+                            build_norm_layer(self.norm_cfg,
+                                             num_channels_cur_layer[i])[1],
+                            nn.ReLU(inplace=True)))
+                else:
+                    transition_layers.append(None)
+            else:
+                conv3x3s = []
+                for j in range(i + 1 - num_branches_pre):
+                    inchannels = num_channels_pre_layer[-1]
+                    outchannels = (
+                        num_channels_cur_layer[i] if j == i -
+                        num_branches_pre else inchannels)
+                    conv3x3s.append(
+                        nn.Sequential(
+                            build_conv_layer(
+                                self.conv_cfg,
+                                inchannels,
+                                outchannels,
+                                3,
+                                2,
+                                1,
+                                bias=False),
+                            build_norm_layer(self.norm_cfg, outchannels)[1],
+                            nn.ReLU(inplace=True)))
+                transition_layers.append(nn.Sequential(*conv3x3s))
+
+        return nn.ModuleList(transition_layers)
+
+    def _make_layer(
+        self,
+        block,
+        inplanes,
+        planes,
+        blocks,
+        num_heads=1,
+        stride=1,
+        window_size=7,
+        mlp_ratio=4.0,
+    ):
+        """Make each layer."""
+        downsample = None
+        if stride != 1 or inplanes != planes * block.expansion:
+            downsample = nn.Sequential(
+                build_conv_layer(
+                    self.conv_cfg,
+                    inplanes,
+                    planes * block.expansion,
+                    kernel_size=1,
+                    stride=stride,
+                    bias=False),
+                build_norm_layer(self.norm_cfg, planes * block.expansion)[1],
+            )
+        layers = []
+
+        if isinstance(block, HRFormerBlock):
```

Comment:
I think there may be a bug in this line. Actually, `block` is not an instance of `HRFormerBlock` or `Bottleneck`, so this line will always return ` False`. Maybe we should use `block == HRFormerBlock`.